pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -480,7 +480,9 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
 				Subjects: []*rbacproto.Subject{
 					{
 						Properties: map[string]string{
-							"source.ip": "192.1.2.0/24",
+							"source.ip":        "192.1.2.0/24",
+							"source.namespace": "default",
+							"source.principal": "cluster.local/ns/default/sa/productpage",
```

Comment:
where do you derive the source namespace from? the principal?
If so, why not make the principal a regex field that allows people to do a lot more than just source namespace? e.g., they could restrict the callers to certain domains (`foo.com/ns/*/sa/*`, or certain classes of namespaces, etc..

Reply:
That would be better, but currently the `ServiceRoleBinding.Subject.user` only supports a native string field using "*" character to indicate if it's a prefix or suffix match (if no "*" then it defaults to exact match).

Updating it to support regex natively could go into the Istio authorization API v2 revamp, @liminw  WDYT?

Reply:
This is neither a prefix/suffix match.. It is a mix of both. @ramaraochavali - this is something similar to the Envoy issue we had - where we wanted the subject alternate name to have some special sematics other than a dumb string compare.. We want to allow all traffic from a particular cluster domain, or from a particular namespace or from a particular service across all clusters, etc..

Reply:
Yeah. Thanks for pointing this. Will look in to this.